dll/win32/shell32/folders/CDesktopFolder.cpp

```diff
@@ -790,13 +813,42 @@ HRESULT WINAPI CDesktopFolder::GetDetailsOf(
         psd->cxChar = DesktopSFHeader[iColumn].cxChar;
         return SHSetStrRet(&psd->str, DesktopSFHeader[iColumn].colnameid);
     }
+    else
+    {   // iColumn is the listview column number in explorer
+        switch (iColumn)
+        {
+            case 0:                /* name */
+                break;
+            case 1:                /* comments */
+                if (_ILIsSpecialFolder(pidl))
+                    iColumnModified = 2; 
+                else
+                    iColumnModified = 5;
+                break;
+            case 2:                /* type */
+                iColumnModified = 1;
+                break;
+            case 3:                /* size */
+                if (_ILIsSpecialFolder(pidl))
+                    iColumnModified = 5;
+                else
+                    iColumnModified = 2;
+                break;
+            case 4:                /* modified date */
+                iColumnModified = 3;
+                break;
+            case 5:                /* attributes */
+                iColumnModified = 4;
+                break;
```

Comment:
What about when iColumn is >= 6 ? (missing "default:" case).
Note that most of the iColumnModified seem to be == iColumn - 1 ; maybe this could be used to simplify a bit further the code, if it makes sense?

Reply:
By removing the 'Comments' column from the display (This matches WinXP and W2K3sp2) the code was very much simplified in this area. Now the iColumn essentially needs no modification. Thanks.

Reply:
> By removing the 'Comments' column from the display (This matches WinXP and W2K3sp2)

Is that a Vista+ column?
Note:
`shell32` is built as Vista and there are Vista builds too.

Reply:
Hardcoding these column numbers just leads to bugs. The shvheader for each IShellFolder implementation should have a matching enum with COL_XYZ (IMHO).

Reply:
I would like to leave adding any enum's for you to do in another PR. Thanks.

Reply:
@HBelusca, now the comments column is restored to the right-most position of the listview. Also, the check for maximum column number is now in file CFSFolder.cpp and functions GetFSColumnDetails and GetDefaultFSColumnState.